File: adventofcode/go/year2020/day15/day15_test.go
package day15

import (
	"testing"

	"github.com/Saser/pdp/adventofcode/go/testcase"
)

var inputFile = testcase.Runfile("adventofcode/data/year2020/day15/actual.in")

var (
	tcPart1 = testcase.NewFile("input", inputFile, "870")
	tcPart2 = testcase.NewFile("input", inputFile, "9136")
)

func TestPart1(t *testing.T) {
	for _, tc := range []testcase.TestCase{
		testcase.New("example1", "0,3,6", "436"),
		testcase.New("example2", "1,3,2", "1"),
		testcase.New("example3", "2,1,3", "10"),
		testcase.New("example4", "1,2,3", "27"),
		testcase.New("example5", "2,3,1", "78"),
		testcase.New("example6", "3,2,1", "438"),
		testcase.New("example7", "3,1,2", "1836"),
		tcPart1,
	} {
		tc.Test(t, Part1)
	}
}

func BenchmarkPart1(b *testing.B) {
	tcPart1.Benchmark(b, Part1)
}

func TestPart2(t *testing.T) {
	for _, tc := range []testcase.TestCase{
		testcase.New("example1", "0,3,6", "175594"),
		testcase.New("example2", "1,3,2", "2578"),
		testcase.New("example3", "2,1,3", "3544142"),
		testcase.New("example4", "1,2,3", "261214"),
		testcase.New("example5", "2,3,1", "6895259"),
		testcase.New("example6", "3,2,1", "18"),
		testcase.New("example7", "3,1,2", "362"),
		tcPart2,
	} {
		tc.Test(t, Part2)
	}
}

func BenchmarkPart2(b *testing.B) {
	tcPart2.Benchmark(b, Part2)
}

File: adventofcode/rust/year2018/day15/day15_test.rs
use adventofcode_rust_aoc as aoc;

mod part1 {
    use super::*;
    aoc::testfn!(
        example1,
        aoc::Data::File("adventofcode/rust/year2018/testdata/day15/ex1"),
        aoc::Data::String("27730"),
        crate::part1
    );
    aoc::testfn!(
        example2,
        aoc::Data::File("adventofcode/rust/year2018/testdata/day15/ex2"),
        aoc::Data::String("36334"),
        crate::part1
    );
    aoc::testfn!(
        example3,
        aoc::Data::File("adventofcode/rust/year2018/testdata/day15/ex3"),
        aoc::Data::String("39514"),
        crate::part1
    );
    aoc::testfn!(
        example4,
        aoc::Data::File("adventofcode/rust/year2018/testdata/day15/ex4"),
        aoc::Data::String("27755"),
        crate::part1
    );
    aoc::testfn!(
        example5,
        aoc::Data::File("adventofcode/rust/year2018/testdata/day15/ex5"),
        aoc::Data::String("28944"),
        crate::part1
    );
    aoc::testfn!(
        example6,
        aoc::Data::File("adventofcode/rust/year2018/testdata/day15/ex6"),
        aoc::Data::String("18740"),
        crate::part1
    );
    aoc::testfn!(
        actual,
        aoc::Data::File("adventofcode/data/year2018/day15/actual.in"),
        aoc::Data::String("201638"),
        crate::part1
    );
}

mod part2 {
    use super::*;
    aoc::testfn!(
        example1,
        aoc::Data::File("adventofcode/rust/year2018/testdata/day15/ex1"),
        aoc::Data::String("4988"),
        crate::part2
    );
    aoc::testfn!(
        example3,
        aoc::Data::File("adventofcode/rust/year2018/testdata/day15/ex3"),
        aoc::Data::String("31284"),
        crate::part2
    );
    aoc::testfn!(
        example4,
        aoc::Data::File("adventofcode/rust/year2018/testdata/day15/ex4"),
        aoc::Data::String("3478"),
        crate::part2
    );
    aoc::testfn!(
        example5,
        aoc::Data::File("adventofcode/rust/year2018/testdata/day15/ex5"),
        aoc::Data::String("6474"),
        crate::part2
    );
    aoc::testfn!(
        example6,
        aoc::Data::File("adventofcode/rust/year2018/testdata/day15/ex6"),
        aoc::Data::String("1140"),
        crate::part2
    );
    aoc::testfn!(
        actual,
        aoc::Data::File("adventofcode/data/year2018/day15/actual.in"),
        aoc::Data::String("95764"),
        crate::part2
    );
}
